lib/chef/knife/bootstrap/train_connector.rb

```diff
@@ -117,7 +117,7 @@ def temp_dir
           cmd = windows? ? MKTEMP_WIN_COMMAND : MKTEMP_NIX_COMMAND
           @tmpdir ||= begin
                         res = run_command!(cmd)
-                        dir = res.stdout.chomp.strip
+                        dir = res.stdout.split.last
```

Comment:
A small comment explaining that we split in case extra output is there would be helpful here. The test captures that scenario, but this might be confusing to someone reading the code.

Reply:
Thanks, I've add comment for this.

Reply:
Looks good now. Thanks